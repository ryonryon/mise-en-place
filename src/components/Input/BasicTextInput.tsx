import styled from '@emotion/styled';
import * as React from 'react';

import InputType from './input-type';
import { secondary } from '../../constants/colors';

const StyledInput = styled.input`
  height: 24px;
  width: 70%;
  margin: 16px;
  padding: 4px;
  border: 1px solid ${secondary};
  color: ${secondary};
  border-radius: 4px;

  &::placeholder {
    color: ${secondary};
  }
`;

export interface Props extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'type'> {
  type: InputType;
}

export const BasicTextInput = ({ type, ...rest }: Props) => {
  return <StyledInput type={type} {...rest} />;
};

BasicTextInput.defaultProps = {
  type: InputType.TEXT
};
