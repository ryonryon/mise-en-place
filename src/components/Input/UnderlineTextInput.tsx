import styled from '@emotion/styled';
import * as React from 'react';

import InputType from './input-type';
import { secondary } from '../../constants/colors';

export interface Props extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'type'> {
  type: InputType;
}

const StyledInput = styled.input`
  height: 24px;
  width: 70%;
  margin: 16px;
  padding: 4px;
  border: none;
  border-bottom: 1px solid ${secondary};
  color: ${secondary};

  &::placeholder {
    color: ${secondary};
  }
`;

export const UnderlineTextInput = ({ type, ...rest }: Props) => <StyledInput type="text" {...rest} />;

UnderlineTextInput.defaultProps = {
  type: InputType.TEXT
};
