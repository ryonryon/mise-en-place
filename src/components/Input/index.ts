export { BasicTextInput } from './BasicTextInput';
export { CustomCheckbox } from './CustomCheckbox';
export { CustomRadioButton } from './CustomRadioButton';
export { CustomRadioButtonGroup } from './CustomRadioButtonGroup';
export { DateTextInput } from './DateTextInput';
export { PasswordTextInput } from './PasswordTextInput';
export { UnderlineTextInput } from './UnderlineTextInput';
export { TextInputWithValidation } from './TextInputWithValidation';
