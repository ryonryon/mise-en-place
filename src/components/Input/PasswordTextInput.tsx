import styled from "@emotion/styled";
import * as React from "react";

import { HideIcon } from "./HideIcon";
import { ShowIcon } from "./ShowIcon";
import InputType from "./input-type";
import { secondary } from "../../constants/colors";
import { BasicTextInput } from "./BasicTextInput";

interface Props extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "type"> {
  showHideIcon: boolean;
}

const PasswordDiv = styled.div`
  margin: 16px;

  & > label {
    text-align: right;
    width: 70%;
    display: inline-block;
    cursor: pointer;
  }

  & > input {
    margin-top: 4px;
  }
`;

export const PasswordTextInput = ({ showHideIcon, ...rest }: Props) => {
  const [value, setValue] = React.useState<string>("");
  const [inputType, setInputType] = React.useState<InputType>(InputType.PASSWORD);

  const handleClick = () => {
    setValue(value);
    setInputType(prevValue => (prevValue === InputType.PASSWORD ? InputType.TEXT : InputType.PASSWORD));
  };

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const val = e.target.value;
    setValue(val);
  };

  return (
    <PasswordDiv>
      {showHideIcon ? (
        <label htmlFor="password-input" onClick={handleClick}>
          {inputType === InputType.PASSWORD ? <HideIcon /> : <ShowIcon />}
        </label>
      ) : null}
      <BasicTextInput type={inputType} name="password-input" {...rest} onChange={handleChange} value={value} />
    </PasswordDiv>
  );
};

PasswordTextInput.defaultProps = {
  showHideIcon: false
};
