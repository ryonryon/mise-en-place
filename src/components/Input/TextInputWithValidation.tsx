import styled from '@emotion/styled';
import * as React from 'react';

import { BasicTextInput } from './BasicTextInput';
import InputType from './input-type';
import { primary, secondary, danger } from '../../constants/colors';

interface Props extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'type'> {
  type: InputType;
  validations: [Function, string][];
}

const TextWithValidationMessages = styled.div`
  & > input {
    border-color: ${secondary};
    color: ${secondary};

    &::placeholder {
      color: ${secondary};
    }
  }

  & > .errorStyle {
    margin-bottom: 4px;
    border-color: ${danger};
  }

  & > .successStyle {
    border-color: ${primary};
  }

  & > ul {
    width: 70%;
    margin: 0 auto;
    padding-left: 30px;
  }

  & > ul > li {
    list-style: none;
    text-align: left;
    font-size: 14px;
    color: ${danger};
  }
`;

export const TextInputWithValidation = ({ validations, ...rest }: Props) => {
  const [value, setValue] = React.useState('');
  const [errorMessages, setErrorMessages] = React.useState<string[]>([]);
  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    e.preventDefault();
    const val = e.currentTarget.value;
    setValue(val);

    const errors: string[] = [];
    for (let i = 0; i < validations.length; i++) {
      const [validation, errorMessage] = validations[i];
      if (validation(val)) errors.push(errorMessage);
    }

    setErrorMessages(errors);
  };
  return (
    <TextWithValidationMessages>
      <BasicTextInput
        type={InputType.TEXT}
        onChange={handleOnChange}
        value={value}
        className={value.length === 0 ? '' : 0 < errorMessages.length ? 'errorStyle' : 'successStyle'}
        {...rest}
      />
      <ul>
        {errorMessages.map((errorMesaage, i) => (
          <li key={i}>{errorMesaage}</li>
        ))}
      </ul>
    </TextWithValidationMessages>
  );
};

TextInputWithValidation.defaultProps = {
  type: InputType.TEXT,
  validations: []
};
