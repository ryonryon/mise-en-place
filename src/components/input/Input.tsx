import * as React from "react";

import { radioData } from "../constants";
import IntroPage from "../app-components/IntroPage";
import BasicTextInput from "./BasicTextInput";
import CustomRadioButtons from "./CustomRadioButton";
import DateTextInput from "./DateTextInput";
import PasswordTextInput from "./PasswordTextInput";
import UnderlineTextInput from "./UnderlineTextInput";
import Section from "../app-components/Section";
import RadioButtons from "./RadioButtons";

const Inputs = () => {
  return (
    <Section title="Inputs">
      <IntroPage title="Basic Text Input" description="Normal text input">
        <BasicTextInput />
      </IntroPage>

      <IntroPage title="Date Text Input" description="Date type input">
        <DateTextInput />
      </IntroPage>

      <IntroPage
        title="Password Text Input"
        description="Password type input which is able to see the password by clicking icon."
      >
        <PasswordTextInput />
      </IntroPage>

      <IntroPage
        title="Underline Text Input"
        description="Underline text input"
      >
        <UnderlineTextInput />
      </IntroPage>

      <IntroPage title="Radio Buttons" description="Normal radio button group">
        <RadioButtons labels={radioData} />
      </IntroPage>

      <IntroPage
        title="Custom Radio Buttons"
        description="Custom radio button group"
      >
        <CustomRadioButtons labels={radioData} />
      </IntroPage>
    </Section>
  );
};

export default Inputs;
