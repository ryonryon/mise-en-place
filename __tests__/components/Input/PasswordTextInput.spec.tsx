import * as React from "react";
import * as renderer from "react-test-renderer";

import { PasswordTextInput } from "../../../src/components/Input/PasswordTextInput";

describe("PasswordTextInput test", () => {
  it("tests PasswordTextInput renders correctly ", () => {
    const passwordTextInput = renderer.create(<PasswordTextInput />).toJSON();
    expect(passwordTextInput).toMatchSnapshot();
  });
});
