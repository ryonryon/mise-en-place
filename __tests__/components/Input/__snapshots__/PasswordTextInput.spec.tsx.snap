// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordTextInput test tests PasswordTextInput renders correctly  1`] = `
<div
  className="css-cqgv7n"
>
  <input
    className="css-1puhmno"
    name="password-input"
    onChange={[Function]}
    type="password"
    value=""
  />
</div>
`;
